lib/rubocop/cop/style/redundant_regexp_escape.rb

```diff
@@ -63,27 +66,30 @@ def autocorrect(node)
 
         private
 
-        def slash_literal?(node)
-          ['/', '%r/'].include?(node.loc.begin.source)
-        end
-
         def allowed_escape?(node, char, within_character_class)
           # Strictly speaking a few single-letter metachars are currently
           # unnecessary to "escape", e.g. g, i, E, F, but enumerating them is
           # rather difficult, and their behaviour could change over time with
           # different versions of Ruby so that e.g. /\g/ != /g/
           return true if /[[:alnum:]]/.match?(char)
-          return true if ALLOWED_ALWAYS_ESCAPES.include?(char)
+          return true if ALLOWED_ALWAYS_ESCAPES.include?(char) || delimiter?(node, char)
 
-          if char == '/'
-            slash_literal?(node)
-          elsif within_character_class
+          if within_character_class
             ALLOWED_WITHIN_CHAR_CLASS_METACHAR_ESCAPES.include?(char)
           else
             ALLOWED_OUTSIDE_CHAR_CLASS_METACHAR_ESCAPES.include?(char)
           end
         end
 
+        def delimiter?(node, char)
+          delimiters = [
+            node.loc.begin.source.chars.last,
```

Comment:
Looks like we're missing a performance cop against `chars.first / last / slice / []`... I created https://github.com/rubocop-hq/rubocop-performance/issues/139

Reply:
Also wondering if `Range#[]` as an optimized shortcut for `Range#source[]` is worth it (avoid intermediate string)